Mathlib/RingTheory/LaurentSeries.lean

```diff
@@ -33,10 +35,39 @@ abbrev LaurentSeries (R : Type*) [Zero R] :=
   HahnSeries ℤ R
 #align laurent_series LaurentSeries
 
-variable {R : Type u}
+variable {R : Type*}
 
 namespace LaurentSeries
 
+section Zero
+
+variable [Zero R]
+
+-- generalize to any locally finite linear order? see mathlib3 PR #18604
+theorem supp_bdd_below_supp_Pwo (f : ℤ → R) (n : ℤ) (hn : ∀(m : ℤ), m < n → f m = 0) :
+    (Function.support f).IsPwo := by
+  rw [← Set.isWf_iff_isPwo, Set.isWf_iff_no_descending_seq]
+  rintro g hg hsupp
+  refine Set.infinite_range_of_injective (StrictAnti.injective hg) ?_
+  refine (Set.finite_Icc n <| g 0).subset <| Set.range_subset_iff.2 ?_
+  simp_all only [Function.mem_support, ne_eq, Set.mem_Icc]
+  intro k
+  constructor
+  exact Int.not_lt.mp (mt (hn (g k)) (hsupp k))
+  exact (StrictAnti.le_iff_le hg).mpr (Nat.zero_le k)
+
+/-- Construct a Laurent series from any function with support that is bounded below. -/
+def LaurentFromSuppBddBelow (f : ℤ → R) (n : ℤ) (hn : ∀(m : ℤ), m < n → f m = 0) :
```

Comment:
We usually use `Of`, not `From`.
```suggestion
@[simps]
def LaurentSeries.ofForallLtEqZero (f : ℤ → R) (n : ℤ) (hn : ∀(m : ℤ), m < n → f m = 0) :
```
Also,

- there is no `BddBelow` in the assumptions;
- do you plan to have a lemma like "the minimal order of a Laurent series constructed this way is at least `n`"? If no, then it makes sense to move `n` inside `∃ n, ∀ m < n, _` or `f =ᶠ[atBot] 0`.

Reply:
I changed the def to use `Bdd Below`, and added lemmas connecting explicit bounds to the `BddBelow` condition and the order of the resulting series.